pycaret/internal/pycaret_experiment/ts_clustering_experiment.py

```diff
@@ -0,0 +1,172 @@
+        X_train = self.train.drop(self.target_param, axis=1)
+
+        if X_train.empty and self.fe_exogenous is None:
+            return None
+        else:
+            # If X_train is not empty or empty but self.fe_exogenous is provided
+            # Return X_train instead of None, since the index can be used to
+            # generate features using self.fe_exogenous
+            return X_train
+
+    @property
+    def X_test(self):
+        # Use index for y_test (idx 2) to get the data
+        test = self.dataset.loc[self.idx[2], :]
+        X_test = test.drop(self.target_param, axis=1)
+
+        if X_test.empty and self.fe_exogenous is None:
+            return None
+        else:
+            # If X_test is not empty or empty but self.fe_exogenous is provided
+            # Return X_test instead of None, since the index can be used to
+            # generate features using self.fe_exogenous
+            return X_test
+
+    @property
+    def test(self):
+        # Return the y_test indices not X_test indices.
+        # X_test indices are expanded indices for handling FH with gaps.
+        # But if we return X_test indices, then we will get expanded test
+        # indices even for univariate time series without exogenous variables
+        # which would be confusing. Hence, we return y_test indices here and if
+        # we want to get X_test indices, then we use self.X_test directly.
+        # Refer:
+        # https://github.com/sktime/sktime/issues/2598#issuecomment-1203308542
+        # https://github.com/sktime/sktime/blob/4164639e1c521b112711c045d0f7e63013c1e4eb/sktime/forecasting/model_evaluation/_functions.py#L196
+        return self.dataset.loc[self.idx[1], :]
+
+    @property
+    def test_transformed(self):
+        # When transforming the test set, we can and should use all data before that
+        # In time series, the order of arguments and returns may be reversed.
+        all_data = pd.concat(
+            [
+                *_pipeline_transform(
+                    pipeline=self.pipeline_fully_trained,
+                    y=self.y,
+                    X=self.X,
+                )
+            ],
+            axis=1,
+        )
+        # Return the y_test indices not X_test indices.
+        # X_test indices are expanded indices for handling FH with gaps.
+        # But if we return X_test indices, then we will get expanded test
+        # indices even for univariate time series without exogenous variables
+        # which would be confusing. Hence, we return y_test indices here and if
+        # we want to get X_test indices, then we use self.X_test directly.
+        # Refer:
+        # https://github.com/sktime/sktime/issues/2598#issuecomment-1203308542
+        # https://github.com/sktime/sktime/blob/4164639e1c521b112711c045d0f7e63013c1e4eb/sktime/forecasting/model_evaluation/_functions.py#L196
+        return all_data.loc[self.idx[1]]
+
+    @property
+    def y_transformed(self):
```

Comment:
In general, you will not have y values in clustering. This would be consistent with both sktime clustering (see [here](https://github.com/sktime/sktime/blob/cf71dd1e8cafbde646b88e448c341e707271ec65/sktime/clustering/k_medoids.py#L91-L99)) and pycaret's _UnsupervisedExperiment (see recommendation of imports [here](https://github.com/pycaret/pycaret/pull/3309#issuecomment-1429443837))

Reply:
Yes thats correct, I will remove and correct all of the methods , first things first, I want to get the class structure and design in place and then we can dive into building out the appropriate methods